.github/workflows/build.yml

```diff
@@ -47,13 +47,22 @@ jobs:
 
       - name: Run the container.
         run: docker run -d --name=solr -p 8983:8983 geerlingguy/solr:${{ env.exact_version }}
+      
+      - name: Wait for the container to start
+        run: sleep 10
```

Comment:
I added a buffer time to wait for the container to finish starting; now it works on my local setup as well.